src/vos/vos_dtx.c

```diff
@@ -1101,3 +1627,263 @@ vos_dtx_mark_sync(daos_handle_t coh, daos_unit_oid_t oid, daos_epoch_t epoch)
 	vos_obj_release(occ, obj, false);
 	return rc;
 }
+
+/*
+ * \return	+2 if current DTX blob is batched cleanup.
+ *		+1 if current DTX entry has ever been committed.
+ *		Zero if the DTX entry is not committed.
+ *		Negative value if error.
+ */
+static int
+dtx_act_handle_batched_cleanup(struct vos_container *cont,
+			       struct dtx_batched_cleanup_blob *bcb,
+			       struct vos_dtx_act_ent_df *dae_df,
+			       umem_off_t *next_dsb)
+{
+	struct umem_instance		*umm;
+	struct vos_dtx_scm_blob		*dsb;
+	struct vos_dtx_record_df	*rec_df;
+	int				 rc;
+
+	if (daos_is_zero_dti(&dae_df->dae_xid) ||
+	    dae_df->dae_flags & DTX_EF_INVALID)
+		return 1;
+
+	if (dae_df->dae_rec_cnt == 0)
```

Comment:
is this possible?

Reply:
That is possible before new punch model. After that, we need to add some waring message for the invalid case.